idx/memory/memory.go

```diff
@@ -45,9 +45,11 @@ type Tree struct {
 }
 
 type Node struct {
-	Path     string
-	Children []string
-	Leaf     bool
+	Path        string
+	Children    []string
+	Defs        []string
+	Leaf        bool
+	HasChildren bool
 }
```

Comment:
can you add a comment to the Node type here as well as the one in idx.go , explaining their respective purpose. for the reader it's confusing why there are two types, how and why they differ.